app/Http/Controllers/Controller.php

```diff
@@ -29,4 +20,22 @@ protected function jsonResponse($data, $code = 200)
             JSON_UNESCAPED_UNICODE
         );
     }
+
+    /**
+     *
+     * @param array|Collection      $items
+     * @param int   $perPage
+     * @param int  $page
+     * @param array $options
+     *
+     * @return LengthAwarePaginator
+     */
+    public function paginate($items, $perPage = 15, $page = null, $options = [])
+    {
+        $page = $page ?: (\Illuminate\Pagination\Paginator::resolveCurrentPage() ?: 1);
+        $items = $items instanceof \Illuminate\Support\Collection ? $items : \Illuminate\Support\Collection::make($items);
+
+        return new \Illuminate\Pagination\LengthAwarePaginator(array_values($items->forPage($page, $perPage)->toArray()), $items->count(), $perPage, $page, $options);
```

Comment:
this is a class created by `Laravel`, allows to paginate any collection passed to it and takes care of all the details for that